pkg/health/checker/checker.go

```diff
@@ -176,10 +221,42 @@ func publishLatestHealth(inCh <-chan api.KVPairs, quitCh <-chan struct{}, result
 	return errCh
 }
 
+func (h healthChecker) getStatusEndpoints(
+	serviceID string,
+	status manifest.StatusStanza,
+) ([]string, error) {
+	rcIDs, err := h.resourceClient.GetRCIDsForPod(types.PodID(serviceID))
```

Comment:
this function depends on fields that are only set if you called the constructor with more arguments. This means that if you 1) use the old constructor and 2) call `WatchService()` or `Service()` with `useHealthService=true` you're going to get a nil pointer exception!

This is making me think that the shadow checker and the non-shadow checker should probably be different types (not just `healthChecker`) and then their behavior can differ without needing to pass extra arguments like `useHealthChecker` to the functions.

A lot of what they do will be common, you can probably extract those into package level functions pretty easily (I hope). These package level functions would have to take all the struct fields as arguments.